cli/internal/fs/hash/capnp.go

```diff
@@ -320,7 +320,10 @@ func HashLockfilePackages(packages []lockfile.Package) (string, error) {
 			return "", err
 		}
 
-		err = entry.SetVersion(pkg.Version)
+		// We explicitly write Version to match Rust behavior when writing empty strings
```

Comment:
Does this count as a global cache key bump? Or is it more of a bug fix? If it's the former, maybe we should fix this on the Rust side.

Reply:
I don't think we need to bump global cache key. All this change should affect (and I believe this is true since existing unit tests for global hash pass without any updates to the snapshots) are workspaces where there's a dependency with an empty version string. At the moment, I think this is only possible via `"foo": "file:"` so I think this should affect ~0 end users.

> maybe we should fix this on the Rust side

I'm very unsure of how to go about this without forking/upstream PR to the Rust capnp crate in order to copy the behavior of the Go library. Since we'll be deleting the Go side soon, I think if we're going to so something "hacky" it should be on the Go side.

Reply:
Fair enough! If it's a bug fix then that works for me.